### PR TITLE
Blink + dependencies: update 3.0.3 -> 3.2.0

### PR DESCRIPTION
**Pull request by leenaars**
###### Motivation for this change

New version available, with version bump of all the dependencies.

Unsure how to deal with https://github.com/NixOS/nixpkgs/issues/65399 as this is PyQt based... 

###### Things done

- [ ] Tested using sandboxing ([nix.useSandbox](http://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](http://nixos.org/nix/manual/#sec-conf-file) on non-NixOS)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nix-review --run "nix-review wip"`
- [ ] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [ ] Ensured that relevant documentation is up to date
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

###### Notify maintainers

cc @

### CODE REVIEW

**Comment by worldofpeace** [comment]
Either of these approaches should work

```
mkDerivationWith python3.pkgs.buildPythonApplication {

  preFixup = ''
    wrapQtApp $out/bin/$executable
  '';
}

mkDerivationWith python3.pkgs.buildPythonApplication {

  dontWrapQtApps = true;

  makeWrapperArgs = [
    "\${qtWrapperArgs[@]}"
  ];
}
```

The first one happens to produce two wrappers, one from python another for qt.
The second one produces one wrapper.

**Comment by leenaars** [comment]
Thanks, that works. Took the latter option.

File: pkgs/applications/networking/instant-messengers/blink/default.nix [hunk 1]
```diff
@@ -1,22 +1,30 @@
...
+    sha256 = "19rcwr5scw48qnj79q1pysw95fz9h98nyc3161qy2kph5g7dwkc3";
   };
 
   patches = [ ./pythonpath.patch ];
   postPatch = ''
     sed -i 's|@out@|'"''${out}"'|g' blink/resources.py
   '';
 
-  propagatedBuildInputs = with pythonPackages; [ pyqt5_with_qtwebkit cjson sipsimple twisted google_api_python_client ];
+  nativeBuildInputs = with pythonPackages; [ wrapQtAppsHook ];
```
Comment (worldofpeace) [inline pkgs/applications/networking/instant-messengers/blink/default.nix]:
```suggestion
```

the mkDerivaitonWith automatically brings in the hook

File: pkgs/applications/networking/instant-messengers/blink/default.nix [hunk 2]
```diff
@@ -1,22 +1,30 @@
...
+
+  propagatedBuildInputs = with pythonPackages; [ pyqt5_with_qtwebkit
```
Comment (worldofpeace) [inline pkgs/applications/networking/instant-messengers/blink/default.nix]: This code style is generally unfavorable in nixpkgs now.

Please format like
```
propagatedBuildInputs = with pythonPackages; [
  pyqt5_with_qtwebkit
  ...
];
```

File: pkgs/applications/networking/instant-messengers/blink/default.nix [hunk 3]
```diff
@@ -30,8 +38,12 @@ pythonPackages.buildPythonApplication rec {
     categories = "Application;Internet;";
   };
 
+  dontWrapQtApps = true;
+
+  makeWrapperArgs = [ "\${qtWrapperArgs[@]}" ];
+
   postInstall = ''
-    wrapProgram $out/bin/blink \
+    wrapQtApp $out/bin/blink \
```
Comment (worldofpeace) [inline pkgs/applications/networking/instant-messengers/blink/default.nix]: You should drop this completely, suggestion above on how to replace it.

File: pkgs/applications/networking/instant-messengers/blink/default.nix [hunk 4]
```diff
@@ -30,8 +38,12 @@ pythonPackages.buildPythonApplication rec {
     categories = "Application;Internet;";
   };
 
+  dontWrapQtApps = true;
+
+  makeWrapperArgs = [ "\${qtWrapperArgs[@]}" ];
```
Comment (worldofpeace) [inline pkgs/applications/networking/instant-messengers/blink/default.nix]:
```suggestion
  makeWrapperArgs = [ 
    "\${qtWrapperArgs[@]}"
    "--prefix LD_LIBRARY_PATH ":" ${gnutls.out}/lib"
  ];
```

File: pkgs/applications/networking/instant-messengers/blink/default.nix [hunk 5]
```diff
@@ -1,22 +1,30 @@
 { stdenv, fetchdarcs, pythonPackages, libvncserver, zlib
-, gnutls, libvpx, makeDesktopItem }:
+, gnutls, libvpx, makeDesktopItem, mkDerivationWith, wrapQtAppsHook }:
```
Comment (worldofpeace) [inline pkgs/applications/networking/instant-messengers/blink/default.nix]:
```suggestion
, gnutls, libvpx, makeDesktopItem, mkDerivationWith }:
```

**Comment by leenaars** [comment]
@worldofpeace: thanks, I took all you suggestions.

**Comment by aanderse** [comment]
@pSub `blink` is broken on `19.09`. I tried `blink` from `master` and this PR fixes the application. Are you able to backport this to `19.09`?

**Comment by worldofpeace** [comment]
@aanderse I did try backporting this, but I didn't have time to review the python changes because it needs them to build.

**Comment by aanderse** [comment]
@worldofpeace ok, thanks for the info.

**Pull request merged**